app/javascript/mastodon/service_worker/web_push_notifications.js

```diff
@@ -92,6 +92,10 @@ export const handlePush = (event) => {
       options.image     = notification.status && notification.status.media_attachments.length > 0 && notification.status.media_attachments[0].preview_url || undefined;
       options.data      = { access_token, preferred_locale, id: notification.status ? notification.status.id : notification.account.id, url: notification.status ? `/@${notification.account.acct}/${notification.status.id}` : `/@${notification.account.acct}` };
 
+      if (notification.type === 'favourite') {
+        options.data.url = `/@${notification.status.account.acct}/${notification.status.id}`;
```

Comment:
Instead of setting this to an incorrect value on line 93 and then "overriding" it with the correct value on 93, let's remove the `url` from the object on line 93, and then set it in an `else if / else` block here.

Reply:
Done!